ui/client/components/modals/ValidationLabels.tsx

```diff
@@ -1,25 +1,49 @@
 import React from "react"
-import {v4 as uuid4} from "uuid"
+import _ from "lodash"
 import {Validator, withoutDuplications} from "../graph/node-modal/editors/Validators"
 
 type Props = {
   validators: Array<Validator>,
   values: Array<string>,
   additionalClassName?: string,
+  validationLabelInfo?: string,
 }
 
 export default function ValidationLabels(props: Props) {
 
-  const {validators, values, additionalClassName} = props
+  type ValidationErrors = {
+    message: string,
+    description: string,
+  }
+
+  const {validators, values, additionalClassName, validationLabelInfo} = props
+
+  const validationErrors: ValidationErrors[] = withoutDuplications(validators)
+    .filter(v => !v.isValid(...values))
+    .map(validator => ({
+      message: validator.message && validator.message(),
+      description: validator.description && validator.description(),
+    }))
+
+  const isValid: boolean = _.isEmpty(validationErrors)
+
+  const renderErrorLablels = () => validationErrors.map(
+    (validationError, ix) => (
+      <span key={ix} className="validation-label-error" title={validationError.description}>
+        {validationError.message}
+      </span>
+    )
+  )
 
   return (
     <div className={`validation-labels ${additionalClassName}`}>
-      {withoutDuplications(validators).map(validator => validator.isValid(...values) ?
-        null : (
-          <span key={uuid4()} className="validation-label" title={validator.description && validator.description()}>
-            {validator.message()}
-          </span>
-        ))}
+      { isValid ? (
```

Comment:
Here we assume we either have errors or validation info? Is it always the case? E.g. expression may type correctly, but is invalid for other reasone?

Reply:
Ah, yes , that's right, I'll change the behaviour to diplay type info when it's correctly inferred, altogheter with errors

Reply:
@mproch 
https://github.com/TouK/nussknacker/pull/1191/commits/3c27991002c0d4bf98b865bcf54e0a057a258aa5